handler/asset_handler.go

```diff
@@ -5,11 +5,164 @@ import (
+			responseBytes, err := json.Marshal(contract.ErrorResponse{Error: "Invalid UUID"})
+			if err != nil {
+				fmt.Printf("handler:Something went wrong while Marshaling,%s ", err.Error())
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+			w.Write(responseBytes)
+			return
+
+		}
+
+		asset, err := asset.DeleteAsset(r.Context(), Id)
+
+		if err != nil {
+			if err == customerrors.AssetAlreadyDeleted {
+				fmt.Println("handler: Asset Already Deleted")
+
+				w.WriteHeader(http.StatusNotFound)
+				responseBytes, err := json.Marshal(contract.ErrorResponse{Error: "Asset Already Deleted"})
+				if err != nil {
+					fmt.Printf("handler:Something went wrong while Marshaling,%s ", err.Error())
+					w.WriteHeader(http.StatusInternalServerError)
+					return
+				}
+				w.Write(responseBytes)
+				return
+			}
+			if err == customerrors.NoAssetsExist {
+				fmt.Println("handler: No assets exist")
+
+				w.WriteHeader(http.StatusNotFound)
+				responseBytes, err := json.Marshal(contract.ErrorResponse{Error: "no asset found"})
+				if err != nil {
+					fmt.Printf("handler:Something went wrong while Marshaling,%s ", err.Error())
+					w.WriteHeader(http.StatusInternalServerError)
+					return
+				}
+				w.Write(responseBytes)
+				return
+			}
+			fmt.Printf("handler:Error while Searching for Updated Asset, %s", err.Error())
+			responseBytes, _ := json.Marshal(contract.ErrorResponse{Error: "something went wrong"})
+			w.Write(responseBytes)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		assetRep := contract.DomainToContractassets(asset)
+		responseBytes, errMarshal := json.Marshal(assetRep)
+		if errMarshal != nil {
+			fmt.Println("handler:Something went wrong while marshalingh assetRep")
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+		w.Write(responseBytes)
+		return
+
+	}
+}
+
+func UpdateAssetHandler(asset service.AssetService) http.HandlerFunc {
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+
+		vars := mux.Vars(r)
+		Id, errParse := uuid.Parse(vars["Id"])
+
+		if errParse != nil {
+			fmt.Println("handler: Invalid UUID")
+
+			w.WriteHeader(http.StatusNotFound)
+			responseBytes, err := json.Marshal(contract.ErrorResponse{Error: "Invalid UUID"})
+			if err != nil {
+				fmt.Printf("handler:Something went wrong while Marshaling,%s ", err.Error())
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+			w.Write(responseBytes)
+			return
+		}
+		var req contract.UpdateRequest
+		errDecoder := json.NewDecoder(r.Body).Decode(&req)
+		if errDecoder != nil {
+			fmt.Println("handler: Something went wrong", errDecoder.Error())
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		asset, err := asset.UpdateAsset(r.Context(), Id, req)
+
+		if err != nil {
+			if err == customerrors.AssetAlreadyDeleted {
+				fmt.Println("handler: Asset Already Deleted")
```

Comment:
same as above

Reply:
because if someone deleted an asset that means Id is still there we are just updating status and changing it to retried that means if by mistake some wants to update deleted asset then it will through an error i have to add this condition because we have to maintain asset record while othere are just deleting whole row so id got deleted